### DISCUSSION TITLE
Templates - create a map

### DISCUSSION THREAD

**Discussion by magick93** [hofstadter-io/hof#178] [Q&A] [answered]
Hello

I'm trying to do something like the following:

```
  {{ $seen := make(map[string]bool) }}
  {{- range $key, $val := .properties -}}
    {{- if .ref_path }}
      {{- $path := .ref_path }}
      {{- if not $seen[$path] }}
        {{- $seen[$path] = true }}
        {{- $path }}
      {{- end }}
    {{- end }}
  {{- end }}
```

The goal being, to only have distinct values.

Is there a way to achieve this in hof?

**Comment by verdverm** [comment, marked as answer]
I would probably do this in the CUE and create a `uniqProperties` field at the same level.

This was one of the motivating cases for adopting CUE, i.e. moving logic out of the templates

**Reply by magick93** [reply]
Ah, good idea! 

https://cuetorials.com/patterns/unique-list/

**Reply by verdverm** [reply]
In the future, CUE will have a better building block for this: https://cuetorials.com/cueology/futurology/associative-lists/

**Reply by magick93** [reply]
This looks cool. 

Do you know when this might land?

**Reply by verdverm** [reply]
nope, they said on the community call today, to nag them via the specific issues, so that it gets put on the roadmap and tied to a version
